### PR TITLE
2 - vni phase 2 - instance changes

### PR DESCRIPTION
**Pull request by ujjwal-ibm**
## Changes in datasource :

- [x] instance profile
  - [x]   code
  - [x]   test
  - [x]   doc

- [x] instance profiles
  - [x]   code
  - [x]   test
  - [x]   doc
- [x] instance
  - [x]   code
  - [x]   test
  - [x]   doc
- [x] instances
  - [x]   code
  - [x]   test
  - [x]   doc

- [x] InstanceNetworkAttachment
  - [x]   code
  - [x]   test
  - [x]   doc
- [x] InstanceNetworkAttachments
  - [x]   code
  - [x]   test
  - [x]   doc

## Changes in resource :

- [x] instance
  - [x]   code
  - [ ]   test
  - [ ]   doc

- [x] InstanceNetworkAttachment
  - [x]   code
  - [x]   test
  - [x]   doc

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
=== RUN   TestAccIBMIsInstanceNetworkAttachmentDataSourceBasic
--- PASS: TestAccIBMIsInstanceNetworkAttachmentDataSourceBasic (44.49s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     46.799s
```
```
=== RUN   TestAccIBMIsInstanceNetworkAttachmentsDataSourceBasic
--- PASS: TestAccIBMIsInstanceNetworkAttachmentsDataSourceBasic (45.47s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     47.523s
```

```
=== RUN   TestAccIBMISInstancesDataSource_vni
--- PASS: TestAccIBMISInstancesDataSource_vni (65.77s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     68.064s
```

```
=== RUN   TestAccIBMISInstanceProfilesDataSource_basic
--- PASS: TestAccIBMISInstanceProfilesDataSource_basic (46.66s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     48.483s
```
```
=== RUN   TestAccIBMISInstanceProfileDataSource_basic
--- PASS: TestAccIBMISInstanceProfileDataSource_basic (48.40s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     53.762s
```
```
=== RUN   TestAccIBMISInstance_vni
--- PASS: TestAccIBMISInstance_vni (246.69s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     248.398s
```

```
=== RUN   TestAccIBMISInstancesDataSource_vni
--- PASS: TestAccIBMISInstancesDataSource_vni (277.96s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     280.343s
```
```
=== RUN   TestAccIBMISInstanceDataSource_vni
--- PASS: TestAccIBMISInstanceDataSource_vni (285.10s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     289.365s
```
```
=== RUN   TestAccIBMIsInstanceNetworkAttachmentsDataSourceBasic
--- PASS: TestAccIBMIsInstanceNetworkAttachmentsDataSourceBasic (252.11s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     254.722s
```

```
=== RUN   TestAccIBMIsInstanceNetworkAttachmentDataSourceBasic
--- PASS: TestAccIBMIsInstanceNetworkAttachmentDataSourceBasic (249.73s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     257.663s
```

### CODE REVIEW

File: ibm/service/vpc/resource_ibm_is_instance.go [hunk 1]
```diff
@@ -507,10 +510,161 @@ func ResourceIBMISInstance() *schema.Resource {
...
+						"deleted": &schema.Schema{
+							Type:        schema.TypeList,
+							MaxItems:    1,
+							Computed:    true,
+							Description: "If present, this property indicates the referenced resource has been deleted, and providessome supplementary information.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"more_info": &schema.Schema{
+										Type:        schema.TypeString,
+										Required:    true,
```
Comment (deepaksibm) [inline ibm/service/vpc/resource_ibm_is_instance.go:531]: should be computed

File: website/docs/d/is_instance.html.markdown [hunk 2]
```diff
@@ -155,6 +155,36 @@ In addition to all argument reference list, you can access the following attribu
      - `enabled` - (Boolean) Indicates whether the metadata service endpoint will be available to the virtual server instance.
      - `protocol` - (String) The communication protocol to use for the metadata service endpoint.
      - `response_hop_limit` - (Integer) The hop limit (IP time to live) for IP response packets from the metadata service.
+
```
Comment (astha-jain) [inline website/docs/d/is_instance.html.markdown:158]: Missing example for instance with network_attachments?

File: website/docs/d/is_instance_network_attachment.html.markdown [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+Provides a read-only data source to retrieve information about an Instance NetworkAttachment. You can then reference the fields of the data source in other resources within the same configuration by using interpolation syntax.
+
+## Example Usage
+
+```terraform
+data "ibm_is_instance_network_attachment" "example" {
+	instance 			= ibm_is_instance.example.id
+	network_attachment 	= ibm_is_instance.example.primary_network_attachment.0.id
+}
```
Comment (astha-jain) [inline website/docs/d/is_instance_network_attachment.html.markdown:19]: Please fix the formatting

File: examples/ibm-is-ng/main.tf [hunk 4]
```diff
@@ -1458,3 +1458,41 @@ resource "ibm_is_virtual_network_interface_ip" "testacc_vni_reservedip" {
...
+	name       			= "${var.name}-ssh"
+	public_key 			= file("~/.ssh/id_rsa.pub")
+}
+resource "ibm_is_instance_network_attachment" "ina" {
+    instance = ibm_is_instance.ins.id
+    name = "viability-undecided-jalapeno-unbuilt"
+    virtual_network_interface {
+        id = ibm_is_virtual_network_interface.testacc_vni2.id
+    }
+}
```
Comment (astha-jain) [inline examples/ibm-is-ng/main.tf:1484]: fix alignment

Comment (ujjwal-ibm) [reply examples/ibm-is-ng/main.tf:1484]: fixed

File: website/docs/d/is_instance_network_attachment.html.markdown (hunk 3 above)
Comment (ujjwal-ibm) [reply website/docs/d/is_instance_network_attachment.html.markdown:19]: fixed

File: website/docs/r/is_instance_network_attachment.html.markdown [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+	- `crn` - (String) The CRN for this subnet.
+	- `deleted` - (List) If present, this property indicates the referenced resource has been deleted, and providessome supplementary information.
+	Nested schema for **deleted**:
+		- `more_info` - (String) Link to documentation about deleted resources.
+	- `href` - (String) The URL for this subnet.
+	- `id` - (String) The unique identifier for this subnet.
+	- `name` - (String) The name for this subnet. The name is unique across all subnets in the VPC.
+	- `resource_type` - (String) The resource type.
+- `type` - (String) The instance network attachment type.
+
```
Comment (sreekarbvibm) [inline website/docs/r/is_instance_network_attachment.html.markdown:72]: I don't see virtual_network_interface being documented in response. Is this discussed ?

**Review by sreekarbvibm — CHANGES_REQUESTED** [review]

File: website/docs/r/is_instance_network_attachment.html.markdown (hunk 5 above)
Comment (sumitkumartiwari) [reply website/docs/r/is_instance_network_attachment.html.markdown:72]: Please also check the code is exist and only doc is missing..

Comment (ujjwal-ibm) [reply website/docs/r/is_instance_network_attachment.html.markdown:72]: its already in the input attributes

File: website/docs/d/is_instance.html.markdown (hunk 2 above)
Comment (ujjwal-ibm) [reply website/docs/d/is_instance.html.markdown:158]: added

**Review by sumitkumartiwari — APPROVED** [review]

**Pull request merged**